webpack.config.js

```diff
@@ -1,17 +1,150 @@
+const path = require('path');
+const { NamedModulesPlugin, EnvironmentPlugin, ProgressPlugin } = require('webpack');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const CopyWebpackPlugin = require('copy-webpack-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const ScriptExtHtmlWebpackPlugin = require('script-ext-html-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const ManifestPlugin = require('webpack-manifest-plugin');
+const HardSourceWebpackPlugin = require('hard-source-webpack-plugin');
+
+const PROD = process.env.NODE_ENV === 'production';
+const filename = `[name]${PROD ? '.[contenthash:8]' : ''}.js`;
+const resolve = (...pathSegments) => path.resolve(__dirname, ...pathSegments);
+
 module.exports = {
-  devtool: process.env.NODE_ENV === 'production' ? 'none' : 'inline-source-map',
+  // build mode
+  mode: PROD ? 'production' : 'development',
+
+  // base resolution path for files
+  context: __dirname,
+
+  // sourcemaps
+  devtool: PROD ? 'none' : 'cheap-module-eval-source-map',
+
+  // server
+  devServer: {
+    contentBase: resolve('dist'),
+    publicPath: '/',
+    stats: 'minimal',
+    port: 3000,
+    hot: true,
+    writeToDisk: true
+  },
+
+  // entry point
+  entry: {
+    promisePolyfill: 'core-js/features/promise',
+    bootstrap: resolve('src/bootstrap.js'),
+  },
+
+  // output config
+  output: {
+    // base directory for output
+    path: resolve('dist'),
+    publicPath: '/',
+    // filenames use content based hashing
+    filename,
+    chunkFilename: filename
+  },
+
+  resolve: {
+
+  },
+
+  // build rules
   module: {
     rules: [
+      // CSS
```

Comment:
I'm glad to see these CSS configs in here. Seems very helpful!

Reply:
I would love to see us move toward using webpack for CSS when we're using it for JS.